client/components/Charts.js

```diff
@@ -370,17 +370,23 @@ function CreateBarChartAll(operations, barchartId) {
 
     // Category -> {Month -> [Amounts]}
     var map = {};
+
+    // Category -> color
+    let colorMap = {};
+
     // Datekey -> Date
     var dateset = {};
-    for (var i = 0; i < operations.length; i++) {
-        var op = operations[i];
-        var c = store.categoryToLabel(op.categoryId);
+    for (let i = 0, size = operations.length; i < size; i++) {
+        let op = operations[i];
+        let c = store.categoryToLabel(op.categoryId);
         map[c] = map[c] || {};
 
         var dk = datekey(op);
         map[c][dk] = map[c][dk] || [];
         map[c][dk].push(op.amount);
         dateset[dk] = +op.date;
+
+        colorMap[c] = colorMap[c] || store.categoryToColor(op.categoryId);
```

Comment:
Thinking about it, it's linear in the number of effectively used categories, so it shouldn't be an issue.